server-core/service/src/test/groovy/io/infectnet/server/service/user/UserServiceImplTest.groovy

```diff
@@ -89,71 +105,139 @@ class UserServiceImplTest extends Specification {
 
-        when: "a user registers in with invalid token"
+        when: "a user registers with invalid username"
             userService.register(TEST_TOKEN,TEST_EMAIL_1,TEST_USERNAME_1,TEST_PASSWORD_1)
 
         then: "InvalidUserNameException is thrown"
             thrown(InvalidUserNameException)
     }
 
-    def "new user tries to register in with invalid email"(){
+    def "new user tries to register with invalid email"(){
         given: "there is a valid token but a conflicting user in the storage"
             def token = new Token(TEST_TOKEN,TEST_VALID_EXPIRATION_DATE)
             tokenStorage.getTokenByTokenString(TEST_TOKEN) >> Optional.of(token)
             def user = new User(TEST_USERNAME_1,TEST_EMAIL_1,TEST_PASSWORD_1,TEST_REGISTRATION_DATE)
             userStorage.getUserByEmail(TEST_EMAIL_1) >> Optional.of(user)
             userStorage.getUserByUserName(TEST_USERNAME_1) >> Optional.empty()
 
-        when: "a user registers in with invalid token"
+        when: "a user registers with invalid email"
             userService.register(TEST_TOKEN,TEST_EMAIL_1,TEST_USERNAME_1,TEST_PASSWORD_1)
 
         then: "InvalidEmailException is thrown"
             thrown(InvalidEmailException)
     }
 
-    def "new user tries to register in with invalid password"(){
+    def "new user tries to register with invalid password"(){
         given: "there is a valid token but conflicting user in the storage"
             def token = new Token(TEST_TOKEN,TEST_VALID_EXPIRATION_DATE)
             tokenStorage.getTokenByTokenString(TEST_TOKEN) >> Optional.of(token)
             userStorage.getUserByEmail(TEST_EMAIL_1) >> Optional.empty()
             userStorage.getUserByUserName(TEST_USERNAME_1) >> Optional.empty()
 
-        when: "a user registers in with invalid token"
+        when: "a user registers with invalid password"
             userService.register(TEST_TOKEN,TEST_EMAIL_1,TEST_USERNAME_1,INVALID_PASSWORD)
 
         then: "InvalidPasswordException is thrown"
             thrown(InvalidPasswordException)
     }
 
+    def "new user tries to register with invalid username and invalid password"(){
+        given: "there is a valid token but conflicting user in the storage"
+            def token = new Token(TEST_TOKEN,TEST_VALID_EXPIRATION_DATE)
+            tokenStorage.getTokenByTokenString(TEST_TOKEN) >> Optional.of(token)
+            def user = new User(TEST_USERNAME_1,TEST_EMAIL_1,TEST_PASSWORD_1,TEST_REGISTRATION_DATE)
+            userStorage.getUserByEmail(TEST_EMAIL_1) >> Optional.empty()
+            userStorage.getUserByUserName(TEST_USERNAME_1) >> Optional.of(user)
+
+        when: "a user registers with invalid password"
+            userService.register(TEST_TOKEN,TEST_EMAIL_1,TEST_USERNAME_1,INVALID_PASSWORD)
+
+        then: "InvalidUserNameException is thrown with InvalidPasswordException chained to it"
+            def ex = thrown(InvalidUserNameException)
+            ex.getTarget() == USERNAME_TARGET
```

Comment:
Replace all occurrences of getters with property accessors:

~~~~Java
ex.target
~~~~

instead of

~~~~Java
ex.getTarget()
~~~~